xl_auth/user/models.py

```diff
@@ -158,6 +159,43 @@ def __repr__(self):
         return '<PasswordReset({email!r})>'.format(email=self.user.email)
 
 
+class LoginAttempt(SurrogatePK, Model):
+    """A failed login attempt."""
+
+    __tablename__ = 'login_attempts'
+    username = Column(db.String(255), unique=False, nullable=False, index=True)
+    source_ip = Column(db.String(255), unique=False, nullable=False, index=True)
+
+    created_at = Column(db.DateTime, default=datetime.utcnow, nullable=False)
+
+    def __init__(self, username, source_ip):
+        """Create instance."""
+        db.Model.__init__(self, username=username, source_ip=source_ip)
+
+    @staticmethod
+    def too_many_recent_failures_for(username):
+        """Determine if a login attempt is allowed or not."""
+        timeframe = current_app.config['FAILED_LOGIN_TIMEFRAME_MINUTES']
+        from_timestamp = datetime.utcnow() - timedelta(minutes=timeframe)
+        attempts = LoginAttempt.query.filter(LoginAttempt.username == username,
```

Comment:
I think this is not robust; should use query similar to *User.get_by_email*, as it is now I could get near-infinite more slack simply by testing with `Mats.blomdahl@kb.se`, `Mats.blomdahl@kb.Se` and so on. :)

Reply:
Changed to use `ilike()`.